boards/shared/Src/App/App_FaultManager.c

```diff
@@ -0,0 +1,81 @@
+#include "App_CanRx.h"
+#include "App_CanTx.h"
+#include "App_FaultManager.h"
+
+#define _add_critical_fault(PREFIX, FAULT)       \
+    if (PREFIX##_GetPeriodicSignal_##FAULT(can)) \
+        return true;
+
+#define add_critical_fault(PREFIX, FAULT) _add_critical_fault(PREFIX, FAULT)
+
+static void *_fsm_can;
+static void *_dcm_can;
+static void *_pdm_can;
+static void *_dim_can;
+static void *_bms_can;
```

Comment:
Could leave the user to define each pointer type but I feel like that's a bit much. I think I am okay with sacrificing type-checking for ease-of-use so I went with void pointers.